crates/sparrow-batch/src/batch.rs

```diff
@@ -77,6 +78,35 @@ impl Batch {
         })
     }
 
+    /// Updates the up_to_time.
+    pub fn with_up_to_time(self, up_to_time: RowTime) -> Self {
```

Comment:
Any details on how this should be used? Eg., should we have an assertion that this only increases the `up_to_time`? Or should we limit this to `up_to_time >= `self.data.max_time` or anythnig like that in assertions?

Reply:
Added assertion that this increases the existing up_to_time. Probably _could_ allow for the latter, but it's not used in that manner, and I don't think we need to support that.